pytorch_lightning/core/optimizer.py

```diff
@@ -184,30 +227,19 @@ def dis_closure():
 
         if make_optimizer_step is None:
             make_optimizer_step = not self._should_accumulate
-
-        trainer = self._trainer
-        optimizer = self._optimizer
+        else:
+            if self._automatic_optimization:
+                raise MisconfigurationException(
+                    'In automatic optimization, `make_optimizer_step` should be None. '
+                    'This option is enabled only with Trainer(automatic_optimization=False) '
+                    'Currently, all optimizers follow the Trainer(accumulate_grad_batches=x) logic')
```

Comment:
```suggestion
                    'In automatic optimization, `make_optimizer_step` should be None.'
                    ' This option is enabled only with `Trainer(automatic_optimization=False)`'
                    ' Currently, all optimizers follow the `Trainer(accumulate_grad_batches=x)` logic')
```
lets add test for all there exceptions...